src/target/codegen.cc

```diff
@@ -83,17 +89,20 @@ class ModuleSerializer {
 
     for (const auto& group : mod_group_vec_) {
       ICHECK_NE(group.size(), 0) << "Every allocated group must have at least one module";
-      if (!group[0]->IsDSOExportable()) {
+      if (group[0]->IsBinarySerializable()) {
```

Comment:
When we are in export_dso mode, we should first check if it is IsDSOExportable, 

```c++
if (export_dso) {
    if (group[0]->IsDSOExportable()) {
     if (group[0]->IsBinarySerializable()) {
    }
} else {
    CHECK(group[0]->IsBinarySerializable());
}
```

```

Reply:
Thank you for the catch, it is reflected.